americangut/per_sample.py

```diff
@@ -515,3 +577,132 @@ def stage_per_sample_specific_statics(opts, sample_ids):
             result[id_] = "Cannot symlink for statics."
 
     return result
+
+
+def _plot_alpha(sample, alpha_map, alpha_field, group_field='SIMPLE_BODY_SITE',
+                output_dir=None, xlabel=None, fp=None, debug=False):
+    """Generates a distrbution plot for the data
+
+    Parameters
+    ----------
+    sample : str
+        The sample ID to be plotted
+    alpha_map_fp : str, dataframe
+        The filepath of a comma-seperated file or the pandas dataframe where
+        the sample ID are given in the `'#SampleID'` column, a column with
+        the name given by `alpha_field` contains alpha diversity values,
+        and the `group_field` column specifying the groups which should be
+        used to seperate the data for making the distribution plot.
+    alpha_field : str
+        The name of the column in `alpha_map` which includes the alpha
+        diversity values.
+    group_field : str
+        Default is 'SIMPLE_BODY_SITE'. The name of the column in `alpha_map`
+        which provides the grouping for generating distribution plots.
+    output_dir : str
+        The location where the alpha diversity figures should be saved.
+    xlabel : str
+        Text describing the quantity on the x-axis.
+
+    Returns
+    -------
+    If the sample is not included in the the mapping file, a string is returned
+    stating this fact.
+
+    If the sample is present, a matplotlib figure with the alpha diversity
+    distribution and a line indicating the sample value is returned.
+
+    If debug is passed, the following parameters are returned:
+        group : str
+            The value of the `group_field` for the sample
+        group_alpha : ndarray
+            The alpha diversity values assoicated with the group
+        sample_alpha : float
+            The alpha diversity for the sample
+        xlabel : str
+            The label used for the x-axis of the plot.
+
+    Raises
```

Comment:
since this has known conditions where it raises, can you these be caught in the driving block above? an exception can then be noted in the `result` object in `alpha_plot`. not catching the exception will result in all of the samples in the block being evaluated being impacted too, so this would help isolate issues.